src/auth/api_key.rs

```diff
@@ -31,10 +31,10 @@ impl ApiKey {
         };
 
         let mut parts = head.uri.into_parts();
-        parts.path_and_query = Some(PathAndQuery::try_from(s).unwrap());
+        parts.path_and_query = Some(PathAndQuery::try_from(s)?);
 
-        head.uri = Uri::from_parts(parts).unwrap();
-        Request::from_parts(head, body)
+        head.uri = Uri::from_parts(parts)?;
+        Ok(Request::from_parts(head, body))
```

Comment:
The calling context already returns a `crate::auth::Result`, so it seems better not to unwrap if we don't need to. Maybe you have an argument for why unwrap is safe here, in which case I can revert changes to this function.